### PR TITLE
doc: Add historical release notes for 0.21.1

### PR DESCRIPTION
**Pull request by Rspigler**
Add release notes to Github docs to finish up the release announcements.

To Do Still:
Mark as 'latest release' on https://github.com/bitcoin/bitcoin/releases w/ short blurb
Announcement on bitcoin-core-dev
@BitcoinCoreOrg twitter announcement

### CODE REVIEW

**Review by fanquake — APPROVED** [review]
ACK ca78b4df1914790900c76a2885daf7ce3df1026e - looks like it matches https://github.com/bitcoin/bitcoin/blob/0.21/doc/release-notes.md.

File: doc/release-notes/release-notes-0.21.1.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+From Bitcoin Core 0.20.0 onwards, macOS versions earlier than 10.12 are no
+longer supported. Additionally, Bitcoin Core does not yet change appearance
+when macOS "dark mode" is activated.
+
+Notable changes
+===============
+
+Taproot Soft Fork
+-----------------
```
Comment (benthecarman) [inline doc/release-notes/release-notes-0.21.1.md:48]: nit: In https://github.com/bitcoin/bitcoin/blob/0.21/doc/release-notes.md it is `## Taproot Soft Fork` and no line of `-` afterwards

File: doc/release-notes/release-notes-0.21.1.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+All users, businesses, and miners are encouraged to upgrade to this release (or a subsequent compatible release) unless they object to activation of taproot. If taproot is locked in, then upgrading before block 709632 is highly recommended to help enforce taproot’s new rules and to avoid the unlikely case of seeing falsely confirmed transactions.
+
+Miners who want to activate Taproot should preferably use this release to control their signaling. The getblocktemplate RPC results will automatically be updated to signal once the appropriate start has been reached and continue signaling until the timeout occurs or taproot activates. Alternatively, miners may manually start signaling on bit 2 at any time; if taproot activates, they will need to ensure they update their nodes before block 709632 or non-upgraded nodes could cause them to mine on an invalid chain. See the versionbits FAQ for details.
```
Comment (MarcoFalke) [inline doc/release-notes/release-notes-0.21.1.md:62]: Why is the text changed? Why are the links removed?

**Review by maflcko — CHANGES_REQUESTED** [review]

**Comment by MarcoFalke** [comment]
Did a proper 1-1 copy in #21830

**Comment by Rspigler** [comment]
Thanks for comments

**Pull request closed without merging**